### PR TITLE
fix: replace label selector mock server test with IT

### PR DESCRIPTION
**Pull request by csviri**
Note that the former test was doing actually more than it should be. So this is a replacement but also change in scope.

### CODE REVIEW

**Comment by metacosm** [comment]
Is what the previous test was covering covered in other tests?

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/labelselector/LabelSelectorTestCustomResource.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.ShortNames;
+import io.fabric8.kubernetes.model.annotation.Version;
+
+@Group("sample.javaoperatorsdk")
+@Version("v1")
+@ShortNames("lst")
+public class LabelSelectorTestCustomResource
+    extends CustomResource<Void, LabelSelectorTestCustomResourceStatus>
```
Comment (metacosm) [inline operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/labelselector/LabelSelectorTestCustomResource.java:13]: Why create a status class when it's not used anywhere and empty?

**Review by metacosm — CHANGES_REQUESTED** [review]

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/labelselector/LabelSelectorTestCustomResource.java (hunk 1 above)
Comment (csviri) [reply operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/labelselector/LabelSelectorTestCustomResource.java]: Good point, fixed. (I think I had an issue with this in the past when status was void, but not anymore)

**Comment by csviri** [comment]
> Is what the previous test was covering covered in other tests?

Yes it should.

**Review by metacosm — APPROVED** [review]

**Review by metacosm — APPROVED** [review]

**Pull request merged**